consensus/service/src/config/tokens.rs

```diff
@@ -0,0 +1,524 @@
+                token_id: Mob::ID,
+                minimum_fee: Some(Mob::MINIMUM_FEE),
+                allow_any_fee: false,
+            }],
+        }
+    }
+}
+
+impl TokensConfig {
+    /// Get the tokens configuration by loading the tokens.toml/json file.
+    pub fn load_from_path(path: impl AsRef<Path>) -> Result<Self, ConsensusServiceError> {
+        let path = path.as_ref();
+
+        // Read configuration file.
+        let data = fs::read_to_string(path).map_err(|err| {
+            ConsensusServiceError::Configuration(format!("error reading file: {}", err.to_string()))
+        })?;
+
+        // Parse configuration file.
+        let tokens_config: Self = match path.extension().and_then(|ext| ext.to_str()) {
+            None => Err(ConsensusServiceError::Configuration(
+                "failed figuring out file extension".to_owned(),
+            )),
+            Some("toml") => toml::from_str(&data).map_err(|err| {
+                ConsensusServiceError::Configuration(format!("TOML parsing: {:?}", err))
+            }),
+            Some("json") => serde_json::from_str(&data).map_err(|err| {
+                ConsensusServiceError::Configuration(format!("JSON parsing: {:?}", err))
+            }),
+            Some(ext) => Err(ConsensusServiceError::Configuration(format!(
+                "Unrecognized extension '{}'",
+                ext
+            ))),
+        }?;
+
+        tokens_config.validate()?;
+        Ok(tokens_config)
+    }
+
+    /// Validate the tokens configuration.
+    pub fn validate(&self) -> Result<(), ConsensusServiceError> {
+        // Cannot have duplicate configuration for a single token.
+        let unique_token_ids = HashSet::from_iter(self.tokens.iter().map(|token| token.token_id));
+        if unique_token_ids.len() != self.tokens.len() {
+            return Err(ConsensusServiceError::Configuration(
+                "duplicate token configuration found".to_owned(),
+            ));
+        }
+
+        // Must have MOB.
+        if self.get_token_config(&Mob::ID).is_none() {
+            return Err(ConsensusServiceError::Configuration(
+                "MOB token configuration not found".to_owned(),
+            ));
+        }
+
+        // Validate the configuration of each token
+        for token in self.tokens.iter() {
+            // We must have a fee for every configured token.
+            if token.minimum_fee_or_default().is_none() {
+                return Err(ConsensusServiceError::Configuration(format!(
+                    "missing minimum fee for token id {:?}",
+                    token.token_id
+                )));
+            }
+
+            // By default, we restrict MOB minimum fee to a sane value.
+            if token.token_id == Mob::ID {
+                let mob_fee = token.minimum_fee_or_default().unwrap(); // We are guaranteed to have a minimum fee for MOB.
+                if !token.allow_any_fee && !(10_000..1_000_000_000_000u64).contains(&mob_fee) {
```

Comment:
Are the endpoints of this range (10_000..1_000_000_000_000u64) constants anywhere? I don't think I've seen that before.